src/write-archive/index.test.ts

```diff
@@ -41,6 +42,76 @@ describe('writeTestResult', () => {
     );
   });
 
+  it('successfully generates test results with mapped source entries', async () => {
+    // @ts-expect-error Jest mock
+    fs.ensureDir.mockReturnValue(true);
+    const oldMockFs = fs.outputFile;
+    const storyJson = {
+      childNodes: [
+        {
+          attributes: {
+            src: '/bogano',
+          },
+        },
+      ],
+    };
+
+    const expectedMappedJson = {
+      childNodes: [
+        {
+          attributes: {
+            src: '/coruscant',
+          },
+        },
+      ],
+    };
+
+    const sourceMapping = new Map<string, string>();
+    sourceMapping.set('/bogano', '/coruscant');
+    await writeTestResult(
+      // the default output directory in playwright
+      { title: 'Toy Story', outputDir: resolve('test-results/toy-story-chromium') } as TestInfo,
+      { home: Buffer.from(JSON.stringify(storyJson)) },
+      {
+        'http://localhost:3000/home': {
+          statusCode: 200,
+          body: Buffer.from(JSON.stringify(storyJson)),
+        },
+      },
+      { viewport: { height: 480, width: 720 } },
+      sourceMapping
+    );
+
+    const expectedBuffer = Buffer.from(JSON.stringify(expectedMappedJson));
+    const oldMock = fs.outputFile;
+
+    // XXX_jwir3: Ugh, this is annoying that I have to do it this way, but I think jest is caching the mocks
+    //            somehow. I always get the wrong result when I use expect(fs.outputFile).toHaveBeenCalledWith().
```

Comment:
What about if you run the test in focussed mode? We have `fs.outputFile.mockClear();` at the top of the test file so it should reset before each individual test.

Reply:
I've refactored this code to work. I have no idea _why_ it didn't work before, but does now, but it's using the same pattern as the other tests with `expect(fs.outputFile).toHaveBeenCalledWith(...)` instead of this crazy hack (that, as @paulelliott pointed out, doesn't even work, anyway).